src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/EndpointInfo/ServerEndpointTrackerV2Tests.cs

```diff
@@ -0,0 +1,168 @@
+            EndpointRemovedEventArgs endpointRemovedArgs = null;
+            tracker.EndpointRemoved += (_, args) => endpointRemovedArgs = args;
+
+            // Act
+            await tracker.PruneEndpointsAsync(cts.Token);
+
+            // Assert
+            IEnumerable<IEndpointInfo> activeEndpoints = await tracker.GetEndpointInfoAsync(cts.Token);
+            Assert.NotEmpty(activeEndpoints);
+
+            Assert.Null(endpointRemovedArgs);
+        }
+
+        [Fact]
+        public async Task PruneEndpointsAsync_DoesNotPrune_TransientlyUnresponsiveEndpoint()
+        {
+            // Arrange
+            ServerEndpointState[] states = [ServerEndpointState.Active, ServerEndpointState.Unresponsive, ServerEndpointState.Active];
+
+            using CancellationTokenSource cts = new CancellationTokenSource(CommonTestTimeouts.GeneralTimeout);
+            using ServerEndpointTrackerV2 tracker = CreateTracker(CreateStateChecker(states));
+
+            IEndpointInfo endpoint = Mock.Of<IEndpointInfo>();
+            await tracker.AddAsync(endpoint, cts.Token);
+
+            EndpointRemovedEventArgs endpointRemovedArgs = null;
+            tracker.EndpointRemoved += (_, args) => endpointRemovedArgs = args;
+
+            // Act & Assert
+            for (int i = 0; i < states.Length; i++)
+            {
+                _timeProvider.Increment(ServerEndpointTrackerV2.UnresponsiveGracePeriod / 2);
+                await tracker.PruneEndpointsAsync(cts.Token);
+
+                IEnumerable<IEndpointInfo> activeEndpoints = await tracker.GetEndpointInfoAsync(cts.Token);
+                IEndpointInfo activeEndpoint = Assert.Single(activeEndpoints);
+                Assert.Equal(endpoint, activeEndpoint);
+            }
+
+            Assert.Null(endpointRemovedArgs);
+        }
+
+        [Fact]
+        public async Task PruneEndpointsAsync_Prunes_UnresponsiveEndpointAfterGracePeriod()
+        {
+            // Arrange
+            using CancellationTokenSource cts = new CancellationTokenSource(CommonTestTimeouts.GeneralTimeout);
+            using ServerEndpointTrackerV2 tracker = CreateTracker(ServerEndpointState.Unresponsive);
+
+            IEndpointInfo endpoint = Mock.Of<IEndpointInfo>();
+            await tracker.AddAsync(endpoint, cts.Token);
+
+            EndpointRemovedEventArgs endpointRemovedArgs = null;
+            tracker.EndpointRemoved += (_, args) => endpointRemovedArgs = args;
+
+            _timeProvider.Increment(ServerEndpointTrackerV2.UnresponsiveGracePeriod * 2);
+
+            // Act
+            await tracker.PruneEndpointsAsync(cts.Token);
+
+            // Assert
+            IEnumerable<IEndpointInfo> activeEndpoints = await tracker.GetEndpointInfoAsync(cts.Token);
+            Assert.Empty(activeEndpoints);
+
+            Assert.NotNull(endpointRemovedArgs);
+            Assert.Equal(ServerEndpointState.Unresponsive, endpointRemovedArgs.State);
+            Assert.Equal(endpointRemovedArgs.Endpoint, endpoint);
```

Comment:
I think these parameters should be switched: `endpoint` is the expected value whereas `endpointRemovedArgs.Endpoint` is the tested value.